tests/integration_tests/test_sim.py

```diff
@@ -84,6 +84,24 @@ def test_config_hello(self, qtbot, sim_app, device_type):
         assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.VERSION_ID) is not None
         assert sim_app.rocket_data.last_value_by_device(device_type, DataEntryIds.DEVICE_TYPE) == device_type
 
+    def test_config_change_gs_address(self, qtbot, sim_app, device_type):
+        flush_packets(sim_app, device_type)
+
+        snapshot = get_event_stats_snapshot()
+
+        sim_app.send_command(device_type.name + ".config")
+        flush_packets(sim_app, device_type)
+
+        assert CONFIG_EVENT.wait(snapshot) == 1
+
+        for connection in sim_app.connections.values():
+            connection._xbee.gs_address = b'\x00\x13\xa2\x00Ag\x8f\xc1'
+
+        sim_app.send_command(device_type.name + ".config")
+        flush_packets(sim_app, device_type)
+
+        assert CONFIG_EVENT.wait(snapshot) == 2
```

Comment:
Can you please double check in the code for `.wait` if a new snapshot needs to be taken before calling this? Alternatively, I think you might be able to call it with `wait(snapshot, num_expected=2)`.

I guess `flush_packets` is delaying it long enough that it doesnt even depend on the waiting, so it'll pass anyways.

Reply:
The code for `.wait` only requires any previously captured snapshot. Resetting `snapshot` here would require the assert to check for a value of 1 both times, but I believe it would make more sense to keep the same snapshot and check for an incremented value after each config event (feel free to correct me if I'm wrong).

Either way, I'll update the second call to `wait()` with the argument `num_expected  = 2`.

Reply:
Sounds good!